r/rsparkling/R/package.R

```diff
@@ -1,13 +1,24 @@
-
 #' @import sparklyr h2o utils
 NULL
 
 # define required spark packages
 spark_dependencies <- function(spark_version, scala_version, ...) {
+  sw_version = getOption("rsparkling.sparklingwater.version", default = "1.6.7")
+
+  if(as.package_version(spark_version)$major != as.package_version(sw_version)$major){
+    stop(cat(paste0("Major version of Sparkling Water does not correspond to major Spark version.
+    \nMajor Sparkling Water Version = ",as.package_version(sw_version)$major,
+    "\nMajor Spark Version = ",as.package_version(spark_version)$major)))
+  }
+  if(as.package_version(spark_version)$minor != as.package_version(sw_version)$minor){
+     stop(cat(paste0("Minor version of Sparkling Water does not correspond to minor Spark version.
+     \nMinor Sparkling Water Version = ",as.package_version(sw_version)$minor,
+     "\nMinor Spark Version = ",as.package_version(spark_version)$minor)))
+  }
   sparklyr::spark_dependency(packages = c(
-    sprintf("ai.h2o:sparkling-water-core_%s:1.6.7", scala_version),
-    sprintf("ai.h2o:sparkling-water-ml_%s:1.6.7", scala_version),
-    sprintf("ai.h2o:sparkling-water-repl_%s:1.6.7", scala_version)
+    sprintf("ai.h2o:sparkling-water-core_%s:%s", scala_version,sw_version),
```

Comment:
We don't check scala_version sparkling-version compatibility - this problem happened to me today - t occurred when testing https://github.com/h2oai/sparkling-water/pull/110

```
:: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
Exception in thread "main" java.lang.RuntimeException: [unresolved dependency: ai.h2o#sparkling-water-core_2.11;1.6.7: not found, unresolved dependency: ai.h2o#sparkling-water-ml_2.11;1.6.7: not found, unresolved dependency: ai.h2o#sparkling-water-repl_2.11;1.6.7: not found]
        at org.apache.spark.deploy.SparkSubmitUtils$.resolveMavenCoordinates(SparkSubmit.scala:1066)
        at org.apache.spark.deploy.SparkSubmit$.prepareSubmitEnvironment(SparkSubmit.scala:294)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:158)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:124)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)```

My `SPARK_HOME` points to spark 2.0 which uses scala 2.11 as default one. The same check is done by sparklyr. So right now it's fetching non-existing package from maven central `sparkling-water-repl_2.11;1.6.7`

We need to check that when user wants to use sparkling water 1.6 and lower, scala 2.10 should be used and scala 2.11 for sparkling water 2.0 - otherwise throw an error.